DistributionUpdate/PackageUpdate/PackageInfoTools.g

```diff
@@ -336,23 +336,34 @@ UpdatePackageInfoFiles := function(pkgdir)
                 "\n  This is not allowed, so the info file will not be changed\n");
           has_error := true;
         fi;
-        if IsBound(infon.Date) then
-          date := infon.Date;
-          if not IsString(date) and Length(date) = 10 and date{[3,6]} = "//" and
-            ForAll( date{ [1,2,4,5,7,8,9,10] }, IsDigitChar ) then
-            Print("  ERROR (", info.PackageName, "): the date ", date,
-                  "should be a string of the form `dd/mm/yyyy'\n" );
-            has_error := true;
-          else
-            date := List( SplitString( date, "/" ), Int);
-            if not date[1] in [1..31] and
-                   date[2] in [1..12] and
-                   date[3] > 1999 and # GAP 4 appeared in 1999
-                   date[1] > DaysInMonth( date[2], date[3] ) then
+        if IsBound(info.Date) then
+          if IsString(infon.date) and Length(infon.Date) = 10 and infon.Date{[3,6]} = "//" and
```

Comment:
Perhaps you can keep the `date := infon.Date;` assignment, and use `date` in a few of these lines -- that would reduce the diff, allowing us to focus on the important bits during review

Reply:
I have no problem with changing it, although the reason why I use `date` and `infon.Date` is that `date` is always a list with which I then check whether the date is valid and `info.Date` is a string storing the new date information. I thought it might be easier to read the code if I separate these. Otherwise the type of `date` would change.